distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/strategy/EvenlySpreadAssignmentStrategy.java

```diff
@@ -0,0 +1,62 @@
+package ru.fix.distributed.job.manager.strategy;
+
+import ru.fix.distributed.job.manager.model.AssignmentState;
+import ru.fix.distributed.job.manager.model.JobId;
+import ru.fix.distributed.job.manager.model.WorkItem;
+import ru.fix.distributed.job.manager.model.WorkerId;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class EvenlySpreadAssignmentStrategy extends AbstractAssignmentStrategy {
+
+    @Override
+    public AssignmentState reassignAndBalance(
+            Map<JobId, Set<WorkerId>> availability,
+            AssignmentState prevAssignment,
+            AssignmentState currentAssignment,
+            Set<WorkItem> itemsToAssign
+    ) {
+        for (Map.Entry<JobId, Set<WorkerId>> jobEntry : availability.entrySet()) {
+            Set<WorkItem> itemsToAssignForJob = getWorkItemsByJob(jobEntry.getKey(), itemsToAssign);
+            jobEntry.getValue().forEach(e -> currentAssignment.putIfAbsent(e, new HashSet<>()));
+
+            int workersCount = jobEntry.getValue().size();
+            int workItemsCount = itemsToAssignForJob.size();
+            int limitWorkItemsOnWorker = limitWorkItemsOnWorker(workItemsCount, workersCount);
+            int itemsAssignedFromPreviousCounter = 0;
+
+            for (WorkItem item : itemsToAssignForJob) {
+                WorkerId workerFromPrevious = prevAssignment.getWorkerOfWorkItem(item);
+
+                if (prevAssignment.containsWorkItem(item)
+                        && currentAssignment.containsKey(workerFromPrevious)
+                        && currentAssignment.isBalanced()
+                        && currentAssignment.isBalancedByJobId(item.getJobId())
+                        && itemsAssignedFromPreviousCounter < limitWorkItemsOnWorker
+                ) {
+                    currentAssignment.addWorkItem(workerFromPrevious, item);
+                    itemsAssignedFromPreviousCounter++;
+
+                    int limitItemsOnWorkerInPreviousState = prevAssignment
+                            .get(workerFromPrevious, jobEntry.getKey()).size();
+                    if (limitItemsOnWorkerInPreviousState <= itemsAssignedFromPreviousCounter) {
+                        itemsAssignedFromPreviousCounter = 0;
+                    }
+                } else {
+                    WorkerId lessBusyWorker = currentAssignment
+                            .getLessBusyWorkerFromAvailableWorkers(jobEntry.getValue());
```

Comment:
evenly spread strategy should  spread work items of job evenly. Suppose that we have w1..w4, w1..w3 have 100 items assigned in currentAssignment from  previous jobs. Now we are processing rebill. getLessBusyWorkerFromAvailableWorkers approach will lead to unevenly spread of rebill work items.